### PR TITLE
Add unique attribute validation for invite flow

### PR DESCRIPTION
**Pull request by thiva-k**
This pull request introduces a new backend executor to enforce uniqueness of user attributes (such as email and username) during the user onboarding flow, and updates both backend and frontend to handle and display attribute conflicts to the user. The changes ensure that users receive immediate feedback if they try to register with an email or username that already exists, improving the overall user experience and data integrity.

https://github.com/user-attachments/assets/3dfa05f2-6e68-4d5f-afea-4e2a19675031

**Backend: Attribute uniqueness validation**

- Added a new executor, `AttributeUniquenessValidator`, which checks if user-supplied values for unique schema attributes (e.g., email, username) already exist in the system. If a conflict is found, the flow is routed back to prompt for a new value, and a user-friendly error is provided. [[1]](diffhunk://#diff-98146efe3858581d419c3403b2895e2e6c900ab1dc5616cf74a3ffd7cc3acd4fR1-R118) [[2]](diffhunk://#diff-82a5330129814e06de3d54d982ddd0ee9e70262be578aa6dcfbe0babfa628d4fR47) [[3]](diffhunk://#diff-0fea251d329deae0b6b8ddab73d999044a47de1d1fc219263dcb7b1125885e06R96-R97)
- Updated the user onboarding flow (`user_onboarding_flow.json`) to insert the uniqueness check after email and user details prompts, ensuring that duplicate values are caught before proceeding. [[1]](diffhunk://#diff-d48c1c0511028fcc9028691ce60290e9db73d16ebb03236c2b8404c699cfaa3aL131-R144) [[2]](diffhunk://#diff-d48c1c0511028fcc9028691ce60290e9db73d16ebb03236c2b8404c699cfaa3aL262-R284)
- Added comprehensive unit tests for the new executor to cover all major scenarios, including conflicts, missing user types, schema errors, and system errors.

**Frontend: User feedback for uniqueness errors**

- Updated the Invite User dialog and Accept Invite box to handle and display flow-level errors, such as attribute uniqueness conflicts, by surfacing the backend-provided failure reason to the user. This gives immediate, clear feedback when a user tries to use a duplicate email or username. [[1]](diffhunk://#diff-7a6dcc7babd1b1b4c9fb1f92a31401c3b593b5227dc644499939cd7d711f0176R78) [[2]](diffhunk://#diff-7a6dcc7babd1b1b4c9fb1f92a31401c3b593b5227dc644499939cd7d711f0176R105) [[3]](diffhunk://#diff-7a6dcc7babd1b1b4c9fb1f92a31401c3b593b5227dc644499939cd7d711f0176L336-L352) [[4]](diffhunk://#diff-7a6dcc7babd1b1b4c9fb1f92a31401c3b593b5227dc644499939cd7d711f0176L486-R481) [[5]](diffhunk://#diff-7a6dcc7babd1b1b4c9fb1f92a31401c3b593b5227dc644499939cd7d711f0176R579) [[6]](diffhunk://#diff-7a6dcc7babd1b1b4c9fb1f92a31401c3b593b5227dc644499939cd7d711f0176R643-R650) [[7]](diffhunk://#diff-db951b8d012b7b5c00bac9e31aacb1e38d22c5ffa4b2339637f004993a203dd3R26) [[8]](diffhunk://#diff-db951b8d012b7b5c00bac9e31aacb1e38d22c5ffa4b2339637f004993a203dd3R63) [[9]](diffhunk://#diff-db951b8d012b7b5c00bac9e31aacb1e38d22c5ffa4b2339637f004993a203dd3R95-R105) [[10]](diffhunk://#diff-db951b8d012b7b5c00bac9e31aacb1e38d22c5ffa4b2339637f004993a203dd3L150-R158)

**Summary of most important changes:**

**Backend: Attribute uniqueness enforcement**
- Added `AttributeUniquenessValidator` executor, integrated it into the executor registry, and updated onboarding flow to check email and user details for uniqueness before proceeding. [[1]](diffhunk://#diff-98146efe3858581d419c3403b2895e2e6c900ab1dc5616cf74a3ffd7cc3acd4fR1-R118) [[2]](diffhunk://#diff-82a5330129814e06de3d54d982ddd0ee9e70262be578aa6dcfbe0babfa628d4fR47) [[3]](diffhunk://#diff-0fea251d329deae0b6b8ddab73d999044a47de1d1fc219263dcb7b1125885e06R96-R97) [[4]](diffhunk://#diff-d48c1c0511028fcc9028691ce60290e9db73d16ebb03236c2b8404c699cfaa3aL131-R144) [[5]](diffhunk://#diff-d48c1c0511028fcc9028691ce60290e9db73d16ebb03236c2b8404c699cfaa3aL262-R284)
- Added thorough unit tests for the new executor to ensure reliability and correct behavior in all edge cases.

**Frontend: User experience improvements**
- Enhanced Invite User dialog and Accept Invite box to display backend-provided flow errors (such as uniqueness conflicts) directly to the user, improving transparency and usability. [[1]](diffhunk://#diff-7a6dcc7babd1b1b4c9fb1f92a31401c3b593b5227dc644499939cd7d711f0176R78) [[2]](diffhunk://#diff-7a6dcc7babd1b1b4c9fb1f92a31401c3b593b5227dc644499939cd7d711f0176R105) [[3]](diffhunk://#diff-7a6dcc7babd1b1b4c9fb1f92a31401c3b593b5227dc644499939cd7d711f0176L336-L352) [[4]](diffhunk://#diff-7a6dcc7babd1b1b4c9fb1f92a31401c3b593b5227dc644499939cd7d711f0176L486-R481) [[5]](diffhunk://#diff-7a6dcc7babd1b1b4c9fb1f92a31401c3b593b5227dc644499939cd7d711f0176R579) [[6]](diffhunk://#diff-7a6dcc7babd1b1b4c9fb1f92a31401c3b593b5227dc644499939cd7d711f0176R643-R650) [[7]](diffhunk://#diff-db951b8d012b7b5c00bac9e31aacb1e38d22c5ffa4b2339637f004993a203dd3R26) [[8]](diffhunk://#diff-db951b8d012b7b5c00bac9e31aacb1e38d22c5ffa4b2339637f004993a203dd3R63) [[9]](diffhunk://#diff-db951b8d012b7b5c00bac9e31aacb1e38d22c5ffa4b2339637f004993a203dd3R95-R105) [[10]](diffhunk://#diff-db951b8d012b7b5c00bac9e31aacb1e38d22c5ffa4b2339637f004993a203dd3L150-R158)

Related Issue:
- https://github.com/asgardeo/thunder/issues/1820

## Summary by CodeRabbit

* **New Features**
  * Onboarding now runs uniqueness checks for email and user details to prevent duplicate accounts; failures route back to the UI for correction.

* **Bug Fixes**
  * Invite and accept dialogs now capture and prioritize flow-specific error messages in their error banners.

* **Tests**
  * Added unit tests covering uniqueness validation, conflict detection, and error/error-path handling.

### CODE REVIEW

File: backend/internal/flow/executor/attribute_uniqueness_validator.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+	if !ok || userType == "" {
+		logger.Debug("User type not found in runtime data, skipping uniqueness check")
+		execResp.Status = common.ExecComplete
+		return execResp, nil
+	}
+
+	svcCtx := security.WithRuntimeContext(context.Background())
+	uniqueAttrs, svcErr := e.userSchemaService.GetUniqueAttributes(svcCtx, userType)
+	if svcErr != nil {
+		logger.Error("Failed to retrieve unique attributes from schema",
```
Comment (ThaminduDilshan) [inline backend/internal/flow/executor/attribute_uniqueness_validator.go]: These are server errors right? For server errors we should return a go error from the execute method. We don't have to log at this level. Task execution node will log and return server error.
ExecFailure is for client errors to be returned with failure reason

File: backend/internal/flow/executor/attribute_uniqueness_validator.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+	execResp := &common.ExecutorResponse{
+		AdditionalData: make(map[string]string),
+		RuntimeData:    make(map[string]string),
+	}
+
+	userType, ok := ctx.RuntimeData[userTypeKey]
+	if !ok || userType == "" {
+		logger.Debug("User type not found in runtime data, skipping uniqueness check")
+		execResp.Status = common.ExecComplete
```
Comment (ThaminduDilshan) [inline backend/internal/flow/executor/attribute_uniqueness_validator.go]: Shouldn't the flow fail in this case? If so we can define usertype as a prerequisite for the executor and use related method from core executor implementation

File: backend/internal/flow/executor/attribute_uniqueness_validator.go (hunk 1 above)
Comment (thiva-k) [reply backend/internal/flow/executor/attribute_uniqueness_validator.go]: Updated

File: backend/internal/flow/executor/attribute_uniqueness_validator.go (hunk 2 above)
Comment (thiva-k) [reply backend/internal/flow/executor/attribute_uniqueness_validator.go]: Updated
